pkg/controller/v1beta1/cluster/reconciler.go

```diff
@@ -111,23 +112,25 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return reconcile.Result{}, err
 	}
 
-	// Reconcile the backend
-	err = r.reconcileBackend(cluster)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-
-	// Reconcile the proxy if configured
-	if cluster.Spec.Proxy != nil {
-		err = r.reconcileProxy(cluster)
+	if cluster.Spec.Backend.Metadata.Name == "" {
```

Comment:
@kuujo 
We can handle this condition differently. We can assume if no name is provided then use default backend reconciler to create deployment, service, etc for current backends like raft. We should think about this part a little bit more

Reply:
Also, I think we can keep this controller as it is with the changes that I have in this PR.